contracts/DaiHrd.sol

```diff
@@ -464,4 +504,99 @@ contract DaiHrd is ERC777 {
 		return attorontodai;
 	}
+
+	// Dai specific functions. These functions all behave similar to standard functions with input or output denominated in Dai instead of DaiHrd
+	function balanceOfDai(address tokenHolder) external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = balanceOf(tokenHolder);
+		return convertAttodaiHrdToAttodai(attodaiHrd, rontodaiPerPot);
+	}
+
+	function totalSupplyDai() external view returns(uint256 attodai) {
+		uint256 rontodaiPerPot = calculatedChi();
+		attodai = convertAttodaiHrdToAttodai(totalSupply(), rontodaiPerPot);
+		return attodai;
+	}
+
+	function withdrawDai(uint256 attodai) external returns(uint256 attodaiHrd) {
+		uint256 rontodaiPerPot = updateAndFetchChi();
+		attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
+		uint256 attodaiWithdrawn = _withdraw(attodaiHrd);
+		require(attodaiWithdrawn > attodai);
+		return attodaiHrd;
+	}
+
+	function sendDenominatedInDai(address recipient, uint256 attodai, bytes calldata data) external {
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
+		_send(_msgSender(), _msgSender(), recipient, attodaiHrd, data, "", true);
+	}
+
+	function burnDenominatedInDai(uint256 attodai, bytes calldata data) external {
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
+		_burn(_msgSender(), _msgSender(), attodaiHrd, data, "");
+	}
+
+	function operatorSendDenominatedInDai(address sender, address recipient, uint256 attodai, bytes calldata data, bytes calldata operatorData) external {
+		require(isOperatorFor(_msgSender(), sender), "ERC777: caller is not an operator for holder");
+
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
+		_send(_msgSender(), sender, recipient, attodaiHrd, data, operatorData, true);
+	}
+
+	function operatorBurnDenominatedInDai(address account, uint256 attodai, bytes calldata data, bytes calldata operatorData) external {
+		require(isOperatorFor(_msgSender(), account), "ERC777: caller is not an operator for holder");
+
+		uint256 rontodaiPerPot = calculatedChi();
+		uint256 attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
+		_burn(_msgSender(), account, attodaiHrd, data, operatorData);
+	}
+
+	// Utility Functions
+	function convertAttodaiToAttodaiHrd(uint256 attodai, uint256 rontodaiPerPot ) internal pure returns (uint256 attodaiHrd) {
+		// + 1 is to compensate rounding, since attodaiHrd is rounded down
+		attodaiHrd = attodai.mul(ONE).div(rontodaiPerPot) + 1;
+		return attodaiHrd;
+	}
+
+	function convertAttodaiHrdToAttodai(uint256 attodaiHrd, uint256 rontodaiPerPot ) internal pure returns (uint256 attodai) {
+		attodai = attodaiHrd.mul(rontodaiPerPot).div(ONE);
+		return attodai;
+	}
+
+	function calculatedChi() internal view returns (uint256 rontodaiPerPot) {
```

Comment:
This could be `public` instead since it is just a `view` function.  May prove useful at some point since Maker doesn't have one.